component/src/main/java/com/celements/photo/utilities/AddAttachmentToDoc.java

```diff
@@ -66,44 +71,34 @@ public XWikiAttachment addAtachment(XWikiDocument doc, ByteArrayOutputStream dat
    * @param context XWikiContext. Needed to get the author and to save the
    *           attachment.
    * @return The attachment containing the given data.
-   * @throws XWikiException
    * 
-   * TODO move to celementsWeb AttachmentService and check that AttachmentEvents
-   *      are fired for lucene! (JIRA: CELDEV-130)
+   * @deprecated since 1.42 instead use {@link IAttachmentServiceRole.addAttachment(
+   *    XWikiDocument, InputStream, String, String, String)}
    */
+  @Deprecated
   public XWikiAttachment addAtachment(XWikiDocument doc, byte[] data, String filename, 
-      XWikiContext context) throws XWikiException{
-    XWikiDocument olddoc = (XWikiDocument) doc.clone();
-    if(olddoc.isNew()) {
-      olddoc = new CreateDocumentCommand().createDocument(olddoc.getDocumentReference(), 
-          "DMS-Document");
-    }
-    XWikiAttachment attachment = olddoc.getAttachment(filename);
-    if (attachment == null) {
-        attachment = new XWikiAttachment();
-        attachment.setDoc(olddoc);
-        attachment.setFilename(filename);
-        olddoc.getAttachmentList().add(attachment);
-    }
-    LOGGER.info("filename='" + filename + "' contentsize='" + data.length + "'");
+      XWikiContext context) throws XWikiException {
     ByteArrayInputStream dataStream = null;
     try {
       dataStream = new ByteArrayInputStream(data);
-      attachment.setContent(dataStream);
-    } catch (IOException ioe) {
-      LOGGER.error("Error setting Attachment content", ioe);
+      return ((AttachmentService) Utils.getComponent(IAttachmentServiceRole.class)
+          ).addAttachment(doc, dataStream, filename, context.getUser(), null);
```

Comment:
why not using new overloaded addAttachment method with byte[] data parameter instead of creating and handling an ByteArrayInputStream in this place?